src/pages/ProfileMappings.tsx

```diff
@@ -0,0 +1,306 @@
+  const appLogo = decodeURIComponent(logo);
+
+  const initialElements = [
+    {
+      id: 'appTitle',
+      sourcePosition: 'right',
+      targetPosition: 'left',
+      type: 'input',
+      data: { label: 'Profile Mappings' },
+      position: { x: 20, y: 200 },
+    },
+    {
+      id: 'appToOkta',
+      sourcePosition: 'right',
+      targetPosition: 'left',
+      type: 'default',
+      data: {
+        label: (
+          <div id="upStream" className="downStream">
+            {label} to Okta user
+          </div>
+        ),
+      },
+      position: { x: 270, y: 80 },
+    },
+    {
+      id: 'oktaToApp',
+      sourcePosition: 'right',
+      targetPosition: 'left',
+      type: 'default',
+      data: {
+        label: (
+          <div id="downStream" className="downStream">
+            Okta user to {label}
+          </div>
+        ),
+      },
+      position: { x: 270, y: 320 },
+      overflow: 'hidden',
+    },
+    { id: 'arrow1', source: 'appTitle', target: `appToOkta`, animated: true },
+    { id: 'arrow2', source: 'appTitle', target: `oktaToApp`, animated: true },
+  ];
+
+  const [attributeMapping, setAttributeMapping] = useState<any>(
+    initialElements
+  );
+
+  const [appToOktaMappingData, setAppToOktaMappingData] = React.useState<any>(
+    {}
+  );
+  const [oktaToAppMappingData, setOktaToAppMappingData] = React.useState<any>(
+    {}
+  );
+  const [loadedData, setLoadedData] = React.useState<boolean>(false);
+
+  const appToOktaApiData = { ...appToOktaMappingData };
+  const oktaToAppApiData = { ...oktaToAppMappingData };
+  const sendUrl = (url: string) => {
+    return request(url);
+  };
+
+  const getApiData = () => {
+    // Authorization: 'SSWS 00Vvsg0QubATDRcwDWxpgRByqcpWJWFGcVjsDQnazX',
+    // const baseUrl = 'https://dev-67150963.okta.com/api/internal/v1/mappings';
+    const url = `/api/internal/v1/mappings`;
+    const appToOktaUrl = `${url}?source=${id2}&target=${id1}`;
+    const oktaToAppUrl = `${url}?source=${id1}&target=${id2}`;
+
+    axios
+      .all([sendUrl(appToOktaUrl), sendUrl(oktaToAppUrl)])
+      .then(
+        axios.spread((...responses) => {
+          let responseOne = responses[0];
+          let responseTwo = responses[1];
+          let appToOktaMappingData = responseOne!.data[0];
+          let oktaToAppMappingData = responseTwo!.data[0];
+          setAppToOktaMappingData(appToOktaMappingData);
+          setOktaToAppMappingData(oktaToAppMappingData);
+          setLoadedData(true);
+        })
+      )
+      .catch((errors) => {
+        console.error(errors);
+      });
+  };
+
+  useEffect(() => {
+    setAttributeMapping(initialElements);
+    getApiData();
+  }, [id2]);
+
+  const onElementClick = (event: any) => {
+    if (event.target.id === 'upStream') {
+      showApptoOktaMapping(appToOktaApiData);
+    }
+    if (event.target.id === 'downStream') {
+      showOktaToAppMapping(oktaToAppApiData);
+    }
+  };
+
```

Comment:
@SaiSrinivasPendyala  i think we can use if and else if 
or we can use an json object and make it to a single line
const targetId={ upStream: appToOktaApiData, downStream: oktaToAppApiData}
showApptoOktaMapping(targetId[event.target.id])
something like this may work

Reply:
Changed it to if - else if with few more changes. @NischalChenna